[inductor] handle non-list/tuple outputs for FallbackKernel

chenyang78 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110145

generate_output may return non-list/tuple outputs. Let's force
those to be list, because we will enumerate kernel.outputs
later in the codegen.

Also fixed a minor issue in an assertion message.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @kadeng @muchulee8 @aakhundov @ColinPeppler

Commit:
[inductor] handle non-list/tuple outputs for FallbackKernel

generate_output may return non-list/tuple outputs. Let's force
those to be list, because we will enumerate kernel.outputs
later in the codegen.

Also fixed a minor issue in an assertion message.

[ghstack-poisoned]

aakhundov commented on torch/_inductor/ir.py:
```diff
@@ -3882,7 +3882,11 @@ def generate_output(output, indices):
                 return None
 
-        packed.outputs = generate_output(example_output, [])
+        outputs = generate_output(example_output, [])
+        if isinstance(outputs, (list, tuple)):
+            packed.outputs = outputs
+        else:
+            packed.outputs = [outputs]
```
Wondering if there is some other code that relies of the `outputs` occasionally not being `(list, tuple)`? I guess, we'll see it in the CI.

chenyang78 commented:
@pytorchbot merge

Commit:
Update on "[inductor] handle non-list/tuple outputs for FallbackKernel"


generate_output may return non-list/tuple outputs. Let's force
those to be list, because we will enumerate kernel.outputs
later in the codegen.

Also fixed a minor issue in an assertion message.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

chenyang78 commented:
@pytorchbot merge
